core/modules/filter/js/filter.js

```diff
@@ -298,6 +272,42 @@ Backdrop.behaviors.editorImageDialog = {
   }
 };
 
+/**
+ * Provides behavior for clicking on images within the library browser.
+ */
+Backdrop.behaviors.editorImageLibrary = {
+  attach: function (context, settings) {
+    // The context may be the image library div itself, so include the context
+    // element in the selector.
+    $('[data-editor-library-view]')
+      .once('editor-library-view')
+      .on('click', '.image-library-choose-file', function() {
+        var $selectedImg = $(this).find('img');
+        var absoluteImgSrc = $selectedImg.data('file-url');
+        var relativeImgSrc = Backdrop.relativeUrl(absoluteImgSrc);
+
+        var $form = $('.filter-format-editor-image-form');
+        $form.find('[name="attributes[src]"]').val(relativeImgSrc);
+        $form.find('[name="fid[fid]"]').val($selectedImg.data('fid'));
+
+        // Reset width and height so image is not stretched to the any
+        // previous image's dimensions.
+        $form.find('[name="attributes[width]"]').val('');
+        $form.find('[name="attributes[height]"]').val('');
+        // Remove style from previous selection.
+        $('.image-library-image-selected').removeClass('image-library-image-selected');
+        // Add style to this selection.
+        $(this).addClass('image-library-image-selected');
+      })
+      .on('dblclick', '.image-library-choose-file', function() {
+        $(this).trigger('click');
+        var $form = $(this).closest('form');
+        var $submit = $form.find('.form-actions input[type=submit]:first');
+        $submit.trigger('mousedown').trigger('click').trigger('mouseup');
+      });
```

Comment:
In testing I realize that we need to use a different mechanism here to find the $form variable, since the view is no longer wrapped by the form but instead a sibling element to it. The double-click behavior does not work currently (should select the image and close the dialog).